tests/libdnf/transaction/MergedTransactionTest.cpp

```diff
@@ -724,19 +724,123 @@ MergedTransactionTest::test_install_downgrade()
         TransactionItemReason::USER
     );
 
+    auto trans3 = std::make_shared< libdnf::swdb_private::Transaction >(conn);
```

Comment:
Why do you use a shared_ptr?

Reply:
Is it because implemented functions in libdnf require it?

Reply:
Yes, the existing code uses shared_ptr already.
This is just a test that uses it internally -> no new public API with a shared pointer -> I don't think it's a problem.